drivers/serial/uart_nrfx_uart.c

```diff
@@ -76,7 +80,8 @@ static struct {
 	volatile size_t rx_counter;
 	volatile size_t rx_offset;
 	int32_t rx_timeout;
-	struct k_delayed_work rx_timeout_work;
+	atomic_t pending_flags[1];
+	struct k_work_delayable rx_timeout_work;
```

Comment:
I think there is no need to involve any `k_work` here, and `k_delayed_work` was simply misused in this case. Just `k_timer` should be sufficient, as it is in `uart_nrfx_uarte.c`. Same applies to TX side. @Mierunski Did I miss some reason why `k_timer` would not work here?

Reply:
If that's the way the Nordic serial folks want to go I'll remove this commit, but the test fails with the current code (#29092 part 2) so it needs to be fixed.  (It'd be a lot simpler if the async API allowed the timeout to be from the first received character, rather than the last received character, as the constant rescheduling likely has a significant impact on ISR time.)